<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Frederic Saab, Peter Kubiniok" />

<meta name="date" content="2021-08-16" />

<title>RHybridFinder</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">RHybridFinder</h1>
<h4 class="author">Frederic Saab, Peter Kubiniok</h4>
<h4 class="date">2021-08-16</h4>



<p>RHybridFinder is a package for the analysis of Mass spectrometry (MS) for the discovery of putative hybrid peptides. For the analysis of your sample, please note that the proposed workflow in the context of this package consists of two major steps:</p>
<ul>
<li>step 1: HybridFinder</li>
<li>Interim: second database search using the merged proteome.</li>
<li>step 2: Either checknetMHCpan or step2_wo_netMHCpan</li>
</ul>
<div id="loading-the-package" class="section level2">
<h2>Loading the package</h2>
<p>After installing the package and in order to be able to use the package, it has to be loaded</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RHybridFinder)</span></code></pre></div>
</div>
<div id="example-data" class="section level2">
<h2>Example data</h2>
<p>For demonstration purposes, the data showcased in this vignette, which is also available in the package (denovo sequencing and database search results, in .csv format) is from HLA Ligand Atlas (Human liver, Autonomous Donor 17) (Marcu et al., 2020). In order to download the human proteome database .fasta file, please visit the <a href="https://www.uniprot.org/uniprot/?query=proteome:UP000005640%20reviewed:yes">uniProt website</a>.</p>
<p>In order to access the example denovo sequencing results and database search results through the package:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#retrieve the denovo sequencing results for the example data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="at">package=</span><span class="st">&quot;RHybridFinder&quot;</span>, denovo_Human_Liver_AUTD17)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#retrieve the database search results for the example data</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="at">package=</span><span class="st">&quot;RHybridFinder&quot;</span>, db_Human_Liver_AUTD17)</span></code></pre></div>
<p>The RAW Mass Spectrometry (MS) Files for the dataset are provided by the authors on Proteomics Identifications Database (PRIDE): PXD019643</p>
</div>
<div id="step-1" class="section level2">
<h2>Step 1</h2>
<p>The step 1 consists of running the HybridFinder function.</p>
<div id="hybridfinder" class="section level3">
<h3>HybridFinder</h3>
<div id="description" class="section level4">
<h4>Description</h4>
<p>The HybridFinder function is based on the workflow proposed by Faridi et al. (2018), with some modifications. Whereby, while using denovo sequencing results with database search results and the proteome database, HybridFinder extracts High confidence denovo peptides and then goes through a 3-step search of these into the proteome. If peptide sequences are matched fully within proteins then they are considered as being “Linear” and Linear peptides within a given spectrum are filtered based on the highest ALC (Average Local Confidence: a significance score for the sequence) score. The rest of the spectra go through the second step during which lists of pair fragments from each peptide sequence are created and then searched in the proteome database, if pair fragments are matched within one protein, these are considered to be potentially cis-spliced. Then, only the highest ALC peptides from each spectrum group are kept. The rest of the spectra goes through the last step, which consists of searching for pair combination matches within two proteins, those that match are considered as being potentially trans-spliced, and only the highest ALC peptides within each spectrum group are kept. Finally, the list of hybrid candidates are concatenated into different ‘fake’ proteins, with the goal being the creation of a hybrid proteome which would mimic the actual proteome. And this hybrid proteome is merged with the reference proteome.</p>
</div>
<div id="loading-data" class="section level4">
<h4>Loading data</h4>
<p>In order to run HybridFinder, three inputs must be provided to <code>HybridFinder</code></p>
<ol style="list-style-type: decimal">
<li>all de novo candidates export file - all denovo sequencing candidates, loaded into R as dataframe</li>
<li>DB search psm export file - database search results, loaded into R as dataframe.</li>
<li>folder path to the proteome database used.</li>
</ol>
<p>Please note that it is recommended to have a folder structure that looks as follows, as it helps keep all results organized:</p>
<ul>
<li>(parent folder)- Exp 1 (could be any name )
<ul>
<li>(child folder): first_run
<ul>
<li>denovo</li>
<li>db</li>
</ul></li>
<li>(2nd child folder): second_run</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>folder_Human_Liver_AUTD17 <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;./data/Human_Liver_AUTD17&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>denovo_Human_Liver_AUTD17 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;first_run&quot;</span>,<span class="st">&quot;all de novo candidates.csv&quot;</span>), <span class="at">sep=</span><span class="st">&quot;,&quot;</span>, <span class="at">head=</span><span class="cn">TRUE</span>,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>db_Human_Liver_AUTD17 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;first_run&quot;</span>,<span class="st">&quot;DB search psm.csv&quot;</span>), <span class="at">sep=</span><span class="st">&quot;,&quot;</span>, <span class="at">head=</span><span class="cn">TRUE</span>,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>proteome_Human_Liver_AUTD17<span class="ot">&lt;-</span> <span class="fu">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;uniprot-proteome-human_UP000005640-reviewed_validated.fasta&quot;</span>)</span></code></pre></div>
</div>
<div id="run-hybridfinder" class="section level4">
<h4>Run HybridFinder</h4>
<p>Once the inputs are loaded, running HybridFinder is a piece of cake. Please note that the HybridFinder function can use parallel computing in order to obtain results fast. It will be good to make sure whether the PC used can support that.</p>
<p>It is possible to set the amount of cores (customCores) for the HybridFinder function to run (given that these are &gt;5). Additionally, it is possible to set a custom ALC cutoff (through the customALCcutoff parameter), setting this allows to filter unassigned spectra based on the newly set custom ALC cutoff (instead of it being calculated). The minimum customALCcutoff score that can be set is 85. Anything set lower than 85 will be set at 85.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>results_HybridFinder_Human_Liver_AUTD17<span class="ot">&lt;-</span> <span class="fu">HybridFinder</span>(<span class="at">denovo_candidates =</span>  denovo_Human_Liver_AUTD17, <span class="at">db_search =</span>  db_Human_Liver_AUTD17, <span class="at">proteome_db =</span> proteome_Human_Liver_AUTD17, <span class="at">customALCcutoff =</span> <span class="cn">NULL</span>, <span class="at">with_parallel =</span> <span class="cn">FALSE</span>, <span class="at">customCores =</span> <span class="dv">8</span>, <span class="at">export_files =</span> <span class="cn">TRUE</span>, <span class="at">export_dir=</span>folder_Human_Liver_AUTD17)</span></code></pre></div>
</div>
<div id="output" class="section level4">
<h4>Output</h4>
<p>The function returns a list composed of 3 elements</p>
<ul>
<li>a dataframe representing the HybridFinder output, containing the high confidence denovo peptides that made it through the different searches (listed above).</li>
<li>a vector containing all the hybrid candidate sequences</li>
<li>a list consisting of the merged reference+hybrid proteome.</li>
</ul>
<div id="ouput-1-hybridfinder-step1-output-hf_step1_output" class="section level5">
<h5>Ouput 1: HybridFinder Step1 output (HF_step1_output)</h5>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#display HybridFinder(HF) step1 output</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">1</span>]]))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Fraction     Scan      m/z    RT    Peptide Length Potential_spliceType ALC</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 394        3  F3:8061 584.3079 45.51 SLVMTQTPKF     10                trans  86</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 23         1 F1:16116 575.7930 72.95  SYLEHLFEL      9               Linear  95</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 375        3 F3:15872 558.8188 70.76 LPVDLATFQL     10                trans  83</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 130        3 F3:13743 533.3030 64.36  SYLLRPVAF      9               Linear  79</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 175        1 F1:11624 531.2999 59.30  YPEDKLLLA      9                  cis  78</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15         1 F1:15777 575.7930 72.95  SYLEHLFEL      9               Linear  92</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                             proteome_database_used</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 394 human_proteome_dropbox_for_HF_validation.fasta</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 23  human_proteome_dropbox_for_HF_validation.fasta</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 375 human_proteome_dropbox_for_HF_validation.fasta</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 130 human_proteome_dropbox_for_HF_validation.fasta</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 175 human_proteome_dropbox_for_HF_validation.fasta</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15  human_proteome_dropbox_for_HF_validation.fasta</span></span></code></pre></div>
</div>
<div id="ouput-2-list-of-step1-candidate-hybrid-peptides" class="section level5">
<h5>Ouput 2: List of step1 candidate hybrid peptides</h5>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#display list of candidate hybrid peptides</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">2</span>]]))</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;YPEDKLLLA&quot; &quot;ANAVLARFY&quot; &quot;YNLPWLENL&quot; &quot;LLYYALMPY&quot; &quot;LPVDLQRYL&quot; &quot;MEDLLKLLA&quot;</span></span></code></pre></div>
</div>
<div id="ouput-3-merged-proteome" class="section level5">
<h5>Ouput 3: merged proteome</h5>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#display the merged proteome</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">tail</span>(results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">3</span>]]))</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $`sp|Q8TDM6-2|DLG5_HUMAN`</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;MEPQRRELLAQCQQSLAQAMTEVEAVLGLLEAAGALSPGERRQLDEEAGGAKAELLLKLLLAKERDHFQDLRAALEKTQPHLLPLLYLNGVVGPPQPAEGAGSTYSVLSTMPSDSESSSSLSSVGTTGKAPSPPPLLTDQQVNEKVENLSLQLRLMTRERNELRKRLAFATHGTAFDKRPYHRLNPDYERLKLQCVRAMSDLQSLQNQHTNALKRCEEVAKETDFYHTLHSRLLSDQTRLKDDVDMLRRENGQLLRERNLLQQSWEDMKRLHEEDQKELGDLRAQQQQVLKHNGSSELLNKLYDTAMDKLEVVKKDYDALRKRYSEKVALHNADLSRLEQLGEENQRLLKQTEMLTQQRDTALQLQHQCALSLRRFEALHHELNKATAQNKDLQWEMELLQSELTELRTTQVKTAKESEKYREERDAVYSEYKLLMSERDQVLSELDKLQTEVELAESKLKSSTSEKKAANEEMEALRQLKDTVTMDAGRANKEVELLRKQCKALCQELKEALQEADVAKCRRDWAFQERDKLVAERDSLRTLCDNLRRERDRAVSELAEALRSLDDTRKQKNDVSRELKELKEQMESQLEKEARFRQLMAHSSHDSALDTDSMEWETEVVEFERETEDLDLKALGFDMAEGVNEPCFPGDCGLFVTKVDKGSLADGRLRVNDWLLRLNDVDLLNKDKKQALKALLNGEGALNMVVRRRKSLGGKVVTPLHLNLSGQKDSGLSLENGVYAAAVLPGSPAAKEGSLAVGDRLVALNGLALDNKSLNECESLLRSCQDSLTLSLLKEQKCVPASGELSPELQEWAPYSPGHSSRHSNPPLYPSRPSVGTVPRSLTPSTTVSSLLRNPLYTVRSHRVGPCSSPPAARDAGPQGLHPSVQHQGRLSLDLSHRTCSDYSEMRATHGSNSLPSSARLGSSSNLQFKAERLKLPSTPRYPRSVVGSERGSVSHSECSTPPQSPLNLDTLSSCSQSQTSASTLPRLAVNPASLGERRKDRPYVEEPRHVKVQKGSEPLGLSLVSGEKGGLYVSKVTVGSLAHQAGLEYGDQLLEFNGLNLRSATEQQARLLLGQQCDTLTLLAQYNPHVHQLSSHSRSSSHLDPAGTHSTLQGSGTTTPEHPSVLDPLMEQDEGPSTPPAKQSSSRLAGDANKKTLEPRVVFLKKSQLELGVHLCGGNLHGVFVAEVEDDSPAKGPDGLVPGDLLLEYGSLDVRNKTVEEVYVEMLKPRDGVRLKVQYRPEEFTKAKGLPGDSFYLRALYDRLADVEQELSFKKDDLLYVDDTLPQGTFGSWMAWQLDENAQKLQRGQLPSKYVMDQEFSRRLSMSEVKDDNSATKTLSAAARRSFFRRKHKHKRSGSKDGKDLLALDAFSSDSLPLFEDSVSLAYQRVQKVDCTALRPVLLLGPLLDVVKEMLVNEAPGKFCRCPLEVMKASQQALERGVKDCLFVDYKRRSGHFDVTTVASLKELTEKNRHCLLDLAPHALERLHHMHLYPLVLFLHYKSAKHLKEQRDPLYLRDKVTQRHSKEQFEAAQKLEQEYSRYFTGVLQGGALSSLCTQLLAMVNQEQNKVLWLPACPL&quot;</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;name&quot;)</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;sp|Q8TDM6-2|DLG5_HUMAN&quot;</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;Annot&quot;)</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;&gt;sp|Q8TDM6-2|DLG5_HUMAN Isoform 2 of Disks large homolog 5 OS=Homo sapiens OX=9606 GN=DLG5&quot;</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;class&quot;)</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;SeqFastaAA&quot;</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $`sp|Q8TDM6-3|DLG5_HUMAN`</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;MEPQRRELLAQCQQSLAQAMTEVEAVLGLLEAAGALSPGERRQLDEEAGGAKAELLLKLLLAKERDHFQDLRAALEKTQPHLLPLLYLNGVVGPPQPAEGAGSTYSVLSTMPSDSESSSSLSSVGTTGKELKEQMESQLEKEARFRQLMAHSSHDSALDTDSMEWETEVVEFERETEDLDLKALGFDMAEGVNEPCFPGDCGLFVTKVDKGSLADGRLRVNDWLLRLNDVDLLNKDKKQALKALLNGEGALNMVVRRRKSLGGKVVTPLHLNLSGQKDSGLSLENGVYAAAVLPGSPAAKEGSLAVGDRLVALNGLALDNKSLNECESLLRSCQDSLTLSLLKVFPQSSSWSGQNLFENLKDSDKMLSFRAHGPEVQAHNKRNLLQHNNSTQTDLFYTDRLEDRKEPGPPGGSSSFLHKPFPGGPLQVCPQACPSASERSLSSFRSDASGDRGFGLVDVRGRRPLLPFETEVGPCGVGEASLDKADSEGSNSGGTWPKAMLSSTAVPEKLSVYKKPKQRKSLFDPNTFKRPQTPPKLDYLLPGPGPAHSPQPSKRAGPLTPPKPPRRSDSLKFQHRLETSSESEATLVGSSPSTSPPSALPPDVDPGEPMHASPPRKARVRLASSYYPEGDGDSSHLPAKKSCDEDLTSQKVDELGQKRRRPKSAPSFRPKLAPVVLPAQFLEV&quot;</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;name&quot;)</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;sp|Q8TDM6-3|DLG5_HUMAN&quot;</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;Annot&quot;)</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;&gt;sp|Q8TDM6-3|DLG5_HUMAN Isoform 3 of Disks large homolog 5 OS=Homo sapiens OX=9606 GN=DLG5&quot;</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;class&quot;)</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;SeqFastaAA&quot;</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $`sp|Q8TDM6-4|DLG5_HUMAN`</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;MPSDSESSSSLSSVGTTGKAPSPPPLLTDQQVNEKVENLSLQLRLMTRERNELRKRLAFATHGTAFDKRPYHRLNPDYERLKLQCVRAMSDLQSLQNQHTNALKRCEEVAKETDFYHTLHSRLLSDQTRLKDDVDMLRRENGQLLRERNLLQQSWEDMKRLHEEDQKELGDLRAQQQQVLKHNGSSELLNKLYDTAMDKLEVVKKDYDALRKRYSEKVALHNADLSRLEQLGEENQRLLKQTEMLTQQRDTALQLQHQCALSLRRFEALHHELNKATAQNKDLQWEMELLQSELTELRTTQVKTAKESEKYREERDAVYSEYKLLMSERDQVLSELDKLQTEVELAESKLKSSTSEKKAANEEMEALRQLKDTVTMDAGRANKEVELLRKQCKALCQELKEALQEADVAKCRRDWAFQERDKLVAERDSLRTLCDNLRRERDRAVSELAEALRSLDDTRKQKNDVSRELKELKEQMESQLEKEARFRQLMAHSSHDSALDTDSMEWETEVVEFERETEDLDLKALGFDMAEGVNEPCFPGDCGLFVTKVDKGSLADGRLRVNDWLLRLNDVDLLNKDKKQALKALLNGEGALNMVVRRRKSLGGKVVTPLHLNLSGQKDSGLSLENGVYAAAVLPGSPAAKEGSLAVGDRLVALNGLALDNKSLNECESLLRSCQDSLTLSLLKVFPQSSSWSGQNLFENLKDSDKMLSFRAHGPEVQAHNKRNLLQHNNSTQTDLFYTDRLEDRKEPGPPGGSSSFLHKPFPGGPLQVCPQACPSASERSLSSFRSDASGDRGFGLVDVRGRRPLLPFETEVGPCGVGEASLDKADSEGSNSGGTWPKAMLSSTAVPEKLSVYKKPKQRKSLFDPNTFKRPQTPPKLDYLLPGPGPAHSPQPSKRAGPLTPPKPPRRSDSLKFQHRLETSSESEATLVGSSPSTSPPSALPPDVDPGEPMHASPPRKARVRLASSYYPEGDGDSSHLPAKKSCDEDLTSQKVDELGQKRRRPKSAPSFRPKLAPVVLPAQFLEEQKCVPASGELSPELQEWAPYSPGHSSRHSNPPLYPSRPSVGTVPRSLTPSTTVSSLLRNPLYTVRSHRVGPCSSPPAARDAGPQGLHPSVQHQGRLSLDLSHRTCSDYSEMRATHGSNSLPSSARLGSSSNLQFKAERLKLPSTPRYPRSVVGSERGSVSHSECSTPPQSPLNLDTLSSCSQSQTSASTLPRLAVNPASLGERRKDRPYVEEPRHVKVQKGSEPLGLSLVSGEKGGLYVSKVTVGSLAHQAGLEYGDQLLEFNGLNLRSATEQQARLLLGQQCDTLTLLAQYNPHVHQLSSHSRSSSHLDPAGTHSTLQGSGTTTPEHPSVLDPLMEQDEGPSTPPAKQSSSRLAGDANKKTLEPRVVFLKKSQLELGVHLCGGNLHGVFVAEVEDDSPAKGPDGLVPGDLLLEYGSLDVRNKTVEEVYVEMLKPRDGVRLKVQYRPEEFTKAKGLPGDSFYLRALYDRLADVEQELSFKKDDLLYVDDTLPQGTFGSWMAWQLDENAQKLQRGQLPSKYVMDQEFSRRLSMSEVKDDNSATKTLSAAARRSFFRRKHKHKRSGSKDGKDLLALDAFSSDSLPLFEDSVSLAYQRVQKVDCTALRPVLLLGPLLDVVKEMLVNEAPGKFCRCPLEVMKASQQALERGVKDCLFVDYKRRSGHFDVTTVASLKELTEKNRHCLLDLAPHALERLHHMHLYPLVLFLHYKSAKHLKEQRDPLYLRDKVTQRHSKEQFEAAQKLEQEYSRYFTGVLQGGALSSLCTQLLAMVNQEQNKVLWLPACPL&quot;</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;name&quot;)</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;sp|Q8TDM6-4|DLG5_HUMAN&quot;</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;Annot&quot;)</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;&gt;sp|Q8TDM6-4|DLG5_HUMAN Isoform 4 of Disks large homolog 5 OS=Homo sapiens OX=9606 GN=DLG5&quot;</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;class&quot;)</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;SeqFastaAA&quot;</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $`sp|Q8TDM6-5|DLG5_HUMAN`</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;MRATHGSNSLPSSARLGSSSNLQFKAERLKLPSTPRYPRSVVGSERGSVSHSECSTPPQSPLNLDTLSSCSQSQTSASTLPRLAVNPASLGERRKDRPYVEEPRHVKVQKGSEPLGLSLVSGEKGGLYVSKVTVGSLAHQAGLEYGDQLLEFNGLNLRSATEQQARLLLGQQCDTLTLLAQYNPHVHQLSSHSRSSSHLDPAGTHSTLQGSGTTTPEHPSVLDPLMEQDEGPSTPPAKQSSSRLAGDANKKTLEPRVVFLKKSQLELGVHLCGGNLHGVFVAEVEDDSPAKGPDGLVPGDLLLEYGSLDVRNKTVEEVYVEMLKPRDGVRLKVQYRPEEFTKAKGLPGDSFYLRALYDRLADVEQELSFKKDDLLYVDDTLPQGTFGSWMAWQLDENAQKLQRGQLPSKYVMDQEFSRRLSMSEVKDDNSATKTLSAAARRSFFRRKHKHKRSGSKDGKDLLALDAFSSDSLPLFEDSVSLAYQRVQKVDCTALRPVLLLGPLLDVVKEMLVNEAPGKFCRCPLEVMKASQQALERGVKDCLFVDYKRRSGHFDVTTVASLKELTEKNRHCLLDLAPHALERLHHMHLYPLVLFLHYKSAKHLKEQRDPLYLRDKVTQRHSKEQFEAAQKLEQEYSRYFTGVLQGGALSSLCTQLLAMVNQEQNKVLWLPACPL&quot;</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;name&quot;)</span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;sp|Q8TDM6-5|DLG5_HUMAN&quot;</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;Annot&quot;)</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;&gt;sp|Q8TDM6-5|DLG5_HUMAN Isoform 5 of Disks large homolog 5 OS=Homo sapiens OX=9606 GN=DLG5&quot;</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;class&quot;)</span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;SeqFastaAA&quot;</span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $`sp|denovo_HF_fake_protein1`</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;YPEDKLLLANYGELFEKFDYGELFEKFDYGELFQKFANAVLARFYKLADFRLLYKLADFLRLYSSYFLLEAFYNLPWLENLLSLNYCLLLLEPFLLPTLPELFLLPTLPLEFLLPTLLTTSWMSLKMEDLLKLLAMENLLKLLALHLSRLQYFLPVDLQRYLLPVNLQRYLLWDLSLTRLFPYYAPELLLLYYASNYRRFLVGSLPKFENGEWRELQLADLFRLYEHVVKVFSLLYEVLLKNFFNLPWTQRF&quot;</span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $`sp|denovo_HF_fake_protein2`</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;FEVPWTQRFDQDLRSMATALLYYASNRYLLYYASRNYLLYYAPLMYLLYYALMPYSLVMTQTPKFYKVYTSVSWMMALLTHGLVLPRRFPQFNYLPWLELNKYPSDEFVFYKPSDEFVFLLYYANSRYLLYYARSNYTVVMTQTPKFTVVMTQTPFKVVYPSMTQRFVVYPAMTQRFRYFSTSVSWYRFSTSVSWLPVDLTAFQLLPVNLATFQLLPVDLATFQLLPVNLKSLTMSYLEHLFYTSYLHELFELANVENLGYLTF&quot;</span></span></code></pre></div>
</div>
</div>
<div id="export" class="section level4">
<h4>Export</h4>
<p>If export is set to <code>TRUE</code> and a valid directory is provided in <code>export_dir</code>, then the results are exported .csv, .csv and .fasta format, respectively.</p>
<p>Even if the export parameters were not set at the beginning, the results returned can always be exported with the <code>export_HybridFinder_results</code> function as long as as the results obtained from the HybridFinder function are stored which is also indicated in the results_list parameter of the <code>export_HybridFinder_results</code> function.</p>
</div>
</div>
</div>
<div id="interim-external-step-second-database-search-using-the-merged-proteome" class="section level2">
<h2>Interim external step: Second database search using the merged proteome</h2>
<p>After finishing this, a second database search has to be done on the raw MS however with the merged proteome (.fasta) exported from the HybridFinder function results.</p>
</div>
<div id="step2" class="section level2">
<h2>Step2</h2>
<p>The second step in RHybridFinder consists of either using <code>checknetMHCpan</code> or <code>step2_wo_netMHCpan</code>, while using the results from step 1 in order to retrieve for the final list of peptides which includes the hybrid candidates, their potential splice types.</p>
<div id="checknetmhcpan" class="section level3">
<h3>checknetMHCpan</h3>
<div id="description-1" class="section level4">
<h4>Description</h4>
<p>the <code>checknetMHCpan</code> function represents step 2 of Faridi et al. (2018)’s workflow and also features the use of netMHCpan (Jurtz et al., 2017, Reynisson et al., 2020) for obtaining the peptide-MHC-I predicted binding affinities. Please note that netMHCpan needs to be installed in order to be able to run this function. The package also contains a function that runs step2 without netMHCpan (Please refer to the <code>step2_wo_netMHCpan</code> part).</p>
</div>
<div id="loading-data-1" class="section level4">
<h4>Loading data</h4>
<p>In order to run checknetMHCpan, four inputs must be provided to <code>checknetMHCpan</code></p>
<ol style="list-style-type: decimal">
<li>netmhcpan_directory: the directory in which netMHCpan is located in (i.e ‘/usr/bin/’ or ‘/usr/bin/local/’)</li>
<li>netmhcpan_alleles: the alleles to be tested again, in a vector format if multiple (i.e alleles&lt;- c(‘HLA-A<em>03:01’, ’HLA-A</em>24:02’))</li>
<li>peptide_rerun: the database search results from the 2nd run loaded into R as a dataframe</li>
<li>HF_step1_output: the dataframe of the first element of the <code>HybridFinder</code> output.</li>
</ol>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>netmhcpan_dir<span class="ot">&lt;-</span> <span class="st">&#39;/usr/bin/&#39;</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>alleles_Human_liver_AUTD17<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;HLA-A*03:01&quot;</span>, <span class="st">&quot;HLA-A*24:02&quot;</span>, <span class="st">&quot;HLA-B*35:03&quot;</span>, <span class="st">&quot;HLA-B*45:01&quot;</span>, <span class="st">&quot;HLA-C*04:01&quot;</span>, <span class="st">&quot;HLA-C*16:01&quot;</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>db_rerun_Human_liver_AUTD17 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;second_run&quot;</span>,<span class="st">&quot;DB search psm.csv&quot;</span>), <span class="at">sep=</span><span class="st">&quot;,&quot;</span>, <span class="at">head=</span><span class="cn">TRUE</span>,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>HF_output_Human_liver_AUTD17<span class="ot">&lt;-</span> results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">1</span>]]</span></code></pre></div>
</div>
<div id="run-checknetmhcpan" class="section level4">
<h4>Run checknetMHCpan</h4>
<p>Once the inputs are loaded, running checknetMHCpan is easier than ABC.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>results_checknetMHCpan_Human_Liver_AUTD17<span class="ot">&lt;-</span> <span class="fu">checknetMHCpan</span>(<span class="at">netmhcpan_directory =</span> netmhcpan_dir, <span class="at">netmhcpan_alleles =</span> alleles_Human_liver_AUTD17, <span class="at">peptide_rerun =</span> db_rerun_Human_liver_AUTD17, <span class="at">HF_step1_output =</span> HF_output_Human_liver_AUTD17, <span class="at">export_files =</span> <span class="cn">TRUE</span>, <span class="at">export_dir=</span>folder_Human_Liver_AUTD17)</span></code></pre></div>
</div>
<div id="output-1" class="section level4">
<h4>Output</h4>
<p>The function returns a list composed of 3 elements: - the netMHCpan results in long format, that is the binding affinity results are displayed for each peptide with a given allele from those chosen. - the netMHCpan results in wide format, that is the binding affinity levels per peptide summarized for all HLA alleles chosen. - the database results with the respective potential splice types retrieved from step 1</p>
<div id="ouput-1-netmhcpan-results-in-long-format" class="section level5">
<h5>Ouput 1: netMHCpan results in long format</h5>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#display netmhcpan output(long version)</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(results_checknetMHCpan_Human_Liver_AUTD17[[<span class="dv">1</span>]]))</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      Pos         HLA      Peptide      Core Of Gp Gl Ip Il        Icore</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 431    1 HLA-A*03:01   VTTYPQGFKL VTTYPQGFK  0  0  0  0  0    VTTYPQGFK</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2846   1 HLA-C*16:01   VTTYPQGFKL VTYPQGFKL  0  1  1  0  0   VTTYPQGFKL</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2789   1 HLA-C*16:01    EYLLKVNEL EYLLKVNEL  0  0  0  0  0    EYLLKVNEL</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 512    1 HLA-A*24:02 GLFEVGAGWLGK GLFAGWLGK  0  3  3  0  0 GLFEVGAGWLGK</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 804    1 HLA-A*24:02   LPVDLATFQL LVDLATFQL  0  1  1  0  0   LPVDLATFQL</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1687   1 HLA-B*45:01    FEVPWTQRF FEVPWTQRF  0  0  0  0  0    FEVPWTQRF</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      Identity     Score Aff(nM)   %Rank  BindLevel strongBinder weakBinder</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 431   PEPLIST 0.2355170  3911.0  4.0177 Non binder                        </span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2846  PEPLIST 0.3118960  1711.5  3.3500 Non binder                        </span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2789  PEPLIST 0.1848720  6764.9  9.4126 Non binder                        </span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 512   PEPLIST 0.0277830 37018.5 36.2242 Non binder                        </span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 804   PEPLIST 0.0686400 23792.1 14.4521 Non binder                        </span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1687  PEPLIST 0.1722820  7752.1  3.4859 Non binder                        </span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       noneBinder Potential_spliceType</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 431  HLA-A*03:01               Linear</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2846 HLA-C*16:01               Linear</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2789 HLA-C*16:01               Linear</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 512  HLA-A*24:02               Linear</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 804  HLA-A*24:02                trans</span></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1687 HLA-B*45:01                trans</span></span></code></pre></div>
</div>
<div id="ouput-2-netmhcpan-results-in-wide-format" class="section level5">
<h5>Ouput 2: netMHCpan results in wide format</h5>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#display netmhcpan output tidied version (wide)</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(results_checknetMHCpan_Human_Liver_AUTD17[[<span class="dv">2</span>]]))</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           Peptide            strongBinder              weakBinder</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 799  GMEGANSLFSGF                                                </span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2311  VPLDEKLLPQL                                                </span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2017    SYNNFFRMF HLA-A*24:02,HLA-C*04:01                        </span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2341   VVYPSMTQRF             HLA-A*24:02 HLA-C*04:01,HLA-C*16:01</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1117   LFVPSYLNLF             HLA-A*24:02             HLA-C*04:01</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 529     FAAFEEPEL HLA-B*35:03,HLA-C*16:01                        </span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                                   noneBinder</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 799  HLA-A*03:01,HLA-A*24:02,HLA-B*35:03,HLA-B*45:01,HLA-C*04:01,HLA-C*16:01</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2311 HLA-A*03:01,HLA-A*24:02,HLA-B*35:03,HLA-B*45:01,HLA-C*04:01,HLA-C*16:01</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2017                         HLA-A*03:01,HLA-B*35:03,HLA-B*45:01,HLA-C*16:01</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2341                                     HLA-A*03:01,HLA-B*35:03,HLA-B*45:01</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1117                         HLA-A*03:01,HLA-B*35:03,HLA-B*45:01,HLA-C*16:01</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 529                          HLA-A*03:01,HLA-A*24:02,HLA-B*45:01,HLA-C*04:01</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      %Rank.HLA-C*16:01 %Rank.HLA-B*45:01 %Rank.HLA-B*35:03 %Rank.HLA-A*03:01</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 799            23.1799            6.9626           27.5516           17.2915</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2311           25.8328           84.5494            2.6277           66.2649</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2017            2.6310           16.6140           20.2119           19.6749</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2341            1.8172           55.9481           15.6904            3.0010</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1117            7.2039           39.4305           13.8101           15.3994</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 529             0.4844           67.7349            0.1084           72.6386</span></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      %Rank.HLA-C*04:01 %Rank.HLA-A*24:02 strongBinder_count weakBinder_count</span></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 799            12.6396            8.4755                  0                0</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2311           23.1188           37.9233                  0                0</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2017            0.3332            0.0133                  2                0</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2341            1.6691            0.0367                  1                2</span></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1117            0.7323            0.0734                  1                1</span></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 529             2.8687           23.6958                  2                0</span></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      noneBinder_count Potential_spliceType</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 799                 6               Linear</span></span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2311                6               Linear</span></span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2017                4               Linear</span></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2341                3                trans</span></span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1117                4               Linear</span></span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 529                 4               Linear</span></span></code></pre></div>
</div>
<div id="ouput-3-database-search-results-updated" class="section level5">
<h5>Ouput 3: Database search results updated</h5>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#display the updated database search results with the categorizations from step1</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(results_checknetMHCpan_Human_Liver_AUTD17[[<span class="dv">3</span>]]))</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                Peptide X.10lgP      Mass Length ppm      m.z Z    RT   Area</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1758       AELLNGKELSA   33.77 1143.6135     11 0.5 572.8143 2 40.00 114240</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4124 AYLERM(+15.99)NYL   32.28 1187.5645      9 0.3 594.7897 2 46.69 600890</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3001 VYTVVDEM(+15.99)F   35.37 1117.5001      9 0.3 559.7575 2 69.18 354900</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3305           YPWTQRF   23.22  996.4818      7 0.5 499.2484 2 57.02      0</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4095         LYQELLHYF   32.28 1224.6179      9 0.5 613.3165 2 70.50  87683</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5386 AYPHNLM(+15.99)TF   24.19 1108.5011      9 0.4 555.2581 2 48.85 426960</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      Fraction    Id     Scan from.Chimera</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1758        3 37741  F3:6305           No</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4124        1  2628  F1:7714           No</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3001        2 25937 F2:15065           No</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3305        1 14824 F1:10950           No</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4095        3 43871 F3:15652           No</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5386        1  3035  F1:8562           No</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                    Source.File</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1758 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4124 171002_AM_BD-ZH17_Liver_W_10%_DDA_#1_400-650mz_msms4.mzML</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3001 171002_AM_BD-ZH17_Liver_W_10%_DDA_#2_400-650mz_msms5.mzML</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3305 171002_AM_BD-ZH17_Liver_W_10%_DDA_#1_400-650mz_msms4.mzML</span></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4095 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5386 171002_AM_BD-ZH17_Liver_W_10%_DDA_#1_400-650mz_msms4.mzML</span></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                        Accession           PTM                   AScore</span></span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1758                      P11586                                       </span></span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4124 P06241:P06241:P06241:P07947 Oxidation (M) M6:Oxidation (M):1000.00</span></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3001                      P53680 Oxidation (M) M8:Oxidation (M):1000.00</span></span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3305                      P68871                                       </span></span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4095                      Q14185                                       </span></span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5386               Q15629:Q15629 Oxidation (M) M7:Oxidation (M):1000.00</span></span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      Found.By Peptide_no_mods Potential_spliceType</span></span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1758 PEAKS DB     AELLNGKELSA               Linear</span></span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4124 PEAKS DB       AYLERMNYL               Linear</span></span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3001 PEAKS DB       VYTVVDEMF               Linear</span></span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3305 PEAKS DB         YPWTQRF               Linear</span></span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4095 PEAKS DB       LYQELLHYF               Linear</span></span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5386 PEAKS DB       AYPHNLMTF               Linear</span></span></code></pre></div>
</div>
</div>
<div id="export-1" class="section level4">
<h4>Export</h4>
<p>If export is set to <code>TRUE</code> and a valid directory is provided in <code>export_dir</code>, then the results are exported .csv, .tsv (tab-separated) and .csv format, respectively.</p>
<p>Even if the export parameters were not set at the beginning, the results returned can always be exported with the <code>export_checknetMHCpan_results</code> function as long as as the results obtained from the checknetMHCpan function are stored which is also indicated in the results_list parameter of the <code>export_checknetMHCpan_results</code> function.</p>
</div>
</div>
<div id="step2_wo_netmhcpan" class="section level3">
<h3>step2_wo_netMHCpan</h3>
<p>The <code>step2_wo_netMHCpan</code>, removes peptide modifications and prepare a peptide (.pep) file for use in webversion of netMHCpan, in case netMHCpan is not installed, OS is windows or the user would like to run in another software. Additionally, the function matches peptide sequences in the database search rerun (the second database search where the merged proteome was used), with the predicted splice type obtained from step 1.</p>
<div id="description-2" class="section level4">
<h4>Description</h4>
<p>The step2_wo_netMHCpan, removes peptide modifications and runs netMHCpan on peptides between 9 and 12-mers. Additionally, the function matches peptide sequences in the database search rerun (the second database search where the merged proteome was used), with predicted splice type obtained from step 1.</p>
</div>
<div id="loading-data-2" class="section level4">
<h4>Loading data</h4>
<p>In order to run checknetMHCpan, four inputs must be provided to <code>checknetMHCpan</code></p>
<ol style="list-style-type: decimal">
<li>peptide_rerun: the database search results from the 2nd run loaded into R as a dataframe</li>
<li>HF_step1_output: the dataframe of the first element of the <code>HybridFinder</code> output.</li>
</ol>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>db_rerun_Human_liver_AUTD17 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;second_run&quot;</span>,<span class="st">&quot;DB search psm.csv&quot;</span>), <span class="at">sep=</span><span class="st">&quot;,&quot;</span>, <span class="at">head=</span><span class="cn">TRUE</span>,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>HF_output_Human_liver_AUTD17<span class="ot">&lt;-</span> results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">1</span>]]</span></code></pre></div>
</div>
<div id="run-step2_wo_netmhcpan" class="section level4">
<h4>Run step2_wo_netMHCpan</h4>
<p>Once the inputs are loaded, running step2_wo_netMHCpan is easier than ABC.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>results_step2_Human_Liver_AUTD17<span class="ot">&lt;-</span> <span class="fu">step2_wo_netMHCpan</span>(<span class="at">peptide_rerun =</span> db_rerun_Human_liver_AUTD17, <span class="at">HF_step1_output =</span> HF_output_Human_liver_AUTD17, <span class="at">export_files =</span> <span class="cn">TRUE</span>, <span class="at">export_dir=</span>folder_Human_Liver_AUTD17)</span></code></pre></div>
</div>
<div id="output-2" class="section level4">
<h4>Output</h4>
<p>The function returns a list composed of 2 elements: - a character vector containing the list of unique peptides from the database search rerun without modifications and of length 9 to 12 amino acids - the database results with the respective potential splice types retrieved from step 1</p>
<div id="ouput-1-netmhcpan-ready-input" class="section level5">
<h5>Ouput 1: netMHCpan-ready input</h5>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#display the netmhcpan-ready input / list of all peptides 9-12 aa, without </span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#modifications</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(results_step2_Human_Liver_AUTD17[[<span class="dv">1</span>]]))</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;LYPDSFTVL&quot;   &quot;LDFPKPLLA&quot;   &quot;YYTPLTPHL&quot;   &quot;LYEPNFLFF&quot;   &quot;VAHVDDMPNAL&quot;</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [6] &quot;FLPLTPQFVTE&quot;</span></span></code></pre></div>
</div>
<div id="ouput-2-database-search-results-updated" class="section level5">
<h5>Ouput 2: Database search results updated</h5>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#display the updated database search results table with the categorizations from </span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co">#step1</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(results_step2_Human_Liver_AUTD17[[<span class="dv">2</span>]]))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          Peptide X.10lgP      Mass Length  ppm      m.z Z    RT    Area</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3078 LRVAPEEHPVL   29.89 1258.7034     11  0.0 420.5751 3 43.15 1350800</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5633   NEQVKNFVA   23.25 1047.5349      9 -0.7 524.7744 2 29.43   65130</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4017   SALPVTYVF   29.62  995.5328      9  0.2 498.7738 2 75.76  101290</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 707    VVYPWTQRF   31.33 1194.6185      9  0.6 598.3169 2 61.47 3779700</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5028   AYLERMNYL   33.65 1171.5696      9  0.6 586.7924 2 51.78   66751</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5789  VVYPAMTQRF   24.10 1210.6168     10 -2.4 606.3142 2 56.05  349750</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      Fraction    Id     Scan from.Chimera</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3078        3 38292  F3:7364           No</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5633        1   322  F1:3088           No</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4017        3 44945 F3:17185           No</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 707         1  5670 F1:13116           No</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5028        3 39990  F3:9972           No</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5789        3 40918 F3:10996           No</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                    Source.File</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3078 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5633 171002_AM_BD-ZH17_Liver_W_10%_DDA_#1_400-650mz_msms4.mzML</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4017 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 707  171002_AM_BD-ZH17_Liver_W_10%_DDA_#1_400-650mz_msms4.mzML</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5028 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5789 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                        Accession PTM AScore Found.By Peptide_no_mods</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3078               P63261:P60709            PEAKS DB     LRVAPEEHPVL</span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5633                      P04114            PEAKS DB       NEQVKNFVA</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4017               Q9BZW5:Q9BZW5            PEAKS DB       SALPVTYVF</span></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 707                       P68871            PEAKS DB       VVYPWTQRF</span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5028 P06241:P06241:P06241:P07947            PEAKS DB       AYLERMNYL</span></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5789    |denovo_HF_fake_protein2            PEAKS DB      VVYPAMTQRF</span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      Potential_spliceType</span></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3078               Linear</span></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5633               Linear</span></span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4017               Linear</span></span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 707                Linear</span></span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5028               Linear</span></span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5789                trans</span></span></code></pre></div>
</div>
</div>
<div id="export-2" class="section level4">
<h4>Export</h4>
<p>If export is set to <code>TRUE</code> and a valid directory is provided in <code>export_dir</code>, then the results are exported .csv, .csv and csv format, respectively.</p>
<p>Even if the export parameters were not set at the beginning, the results returned can always be exported with the <code>export_step2_results</code> function as long as as the results obtained from the step2_wo_netMHCpan function are stored which is also indicated in the results_list parameter of the <code>export_step2_results</code> function.</p>
</div>
</div>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Faridi, P., Li, C., Ramarathinam, S. H., Vivian, J. P., Illing, P. T., Mifsud, N. A., Ayala, R., Song, J., Gearing, L. J., Hertzog, P. J., Ternette, N., Rossjohn, J., Croft, N. P., &amp; Purcell, A. W. (2018). A subset of HLA-I peptides are not genomically templated: Evidence for cis- and trans-spliced peptide ligands. Science Immunology, 3(28), eaar3947. /sciimmunol.aar3947, <a href="https://immunology.sciencemag.org/content/3/28/eaar3947">Link</a></p>
<p>Hanada K, Yewdell JW, Yang JC. Immune recognition of a human renal cancer antigen through post-translational protein splicing. Nature. 2004 Jan 15;427(6971):252-6. DOI /nature02240, <a href="https://www.nature.com/articles/nature02240">Link</a></p>
<p>Marcu A, Bichmann L, Kuchenbecker L, et al HLA Ligand Atlas: a benign reference of HLA-presented peptides to improve T-cell-based cancer immunotherapyJournal for ImmunoTherapy of Cancer 2021;9:e002071. /jitc-2020-002071, <a href="https://jitc.bmj.com/content/9/4/e002071">Link</a></p>
<p>Birkir Reynisson, Bruno Alvarez, Sinu Paul, Bjoern Peters, Morten Nielsen, NetMHCpan-4.1 and NetMHCIIpan-4.0: improved predictions of MHC antigen presentation by concurrent motif deconvolution and integration of MS MHC eluted ligand data, Nucleic Acids Research, Volume 48, Issue W1, 02 July 2020, Pages W449–W454, /nar/gkaa379, <a href="https://academic.oup.com/nar/article/48/W1/W449/5837056">Link</a></p>
<p>Jurtz V, Paul S, Andreatta M, Marcatili P, Peters B, Nielsen M. NetMHCpan-4.0: Improved Peptide-MHC Class I Interaction Predictions Integrating Eluted Ligand and Peptide Binding Affinity Data. J Immunol. 2017 Nov 1;199(9):3360-3368. Epub 2017 Oct 4. PMID: 28978689; PMCID: PMC5679736 /jimmunol.1700893, <a href="https://academic.oup.com/nar/article/48/W1/W449/5837056">Link</a></p>
<p>The UniProt Consortium, UniProt: the universal protein knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1, 8 January 2021, Pages D480–D489, /nar/gkaa1100, <a href="https://academic.oup.com/nar/article/49/D1/D480/6006196">Link</a></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
