<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Frederic Saab, Peter Kubiniok" />

<meta name="date" content="2021-06-18" />

<title>RHybridFinder</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">RHybridFinder</h1>
<h4 class="author">Frederic Saab, Peter Kubiniok</h4>
<h4 class="date">2021-06-18</h4>



<p>RHybridFinder is a package for the analysis of Mass spectrometry (MS) for the discovery of putative hybrid peptides. For the analysis of your sample, please note that the proposed workflow in the context of this package consists of two major steps:</p>
<ul>
<li>step 1: HybridFinder</li>
<li>Interim: second database search using the merged proteome.</li>
<li>step 2: Either checknetMHCpan or step2_wo_netMHCpan</li>
</ul>
<div id="loading-the-package" class="section level2">
<h2>Loading the package</h2>
<p>After installing the package and in order to be able to use the package, it has to be loaded</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(RHybridFinder)</span></code></pre></div>
</div>
<div id="example-data" class="section level2">
<h2>Example data</h2>
<p>For demonstration purposes, the data showcased in this vignette, which is also available in the package (denovo sequencing and database search results, in .csv format) is from HLA Ligand Atlas (Human liver, Autonomous Donor 17) (Marcu et al., 2020). In order to download the human proteome database .fasta file, please visit the <a href="https://www.uniprot.org/uniprot/?query=proteome:UP000005640%20reviewed:yes">uniProt website</a>.</p>
<p>In order to access the example denovo sequencing results and database search results through the package:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co">#retrieve the denovo sequencing results for the example data</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw">data</span>(<span class="dt">package=</span><span class="st">&quot;RHybridFinder&quot;</span>, denovo_Human_Liver_AUTD17)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co">#retrieve the database search results for the example data</span></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="kw">data</span>(<span class="dt">package=</span><span class="st">&quot;RHybridFinder&quot;</span>, db_Human_Liver_AUTD17)</span></code></pre></div>
<p>The RAW Mass Spectrometry (MS) Files for the dataset are provided by the authors on Proteomics Identifications Database (PRIDE): PXD019643</p>
</div>
<div id="step-1" class="section level2">
<h2>Step 1</h2>
<p>The step 1 consists of running the HybridFinder function.</p>
<div id="hybridfinder" class="section level3">
<h3>HybridFinder</h3>
<div id="description" class="section level4">
<h4>Description</h4>
<p>The HybridFinder function is based on the workflow proposed by Faridi et al. (2018), with some modifications. Whereby, while using denovo sequencing results with database search results and the proteome database, HybridFinder extracts High confidence denovo peptides and then goes through a 3-step search of these into the proteome. If peptide sequences are matched fully within proteins then they are considered as being “Linear” and Linear peptides within a given spectrum are filtered based on the highest ALC (Average Local Confidence: a significance score for the sequence) score. The rest of the spectra go through the second step during which lists of pair fragments from each peptide sequence are created and then searched in the proteome database, if pair fragments are matched within one protein, these are considered to be potentially cis-spliced. Then, only the highest ALC peptides from each spectrum group are kept. The rest of the spectra goes through the last step, which consists of searching for pair combination matches within two proteins, those that match are considered as being potentially trans-spliced, and only the highest ALC peptides within each spectrum group are kept. Finally, the list of hybrid candidates are concatenated into different ‘fake’ proteins, with the goal being the creation of a hybrid proteome which would mimic the actual proteome. And this hybrid proteome is merged with the reference proteome.</p>
</div>
<div id="loading-data" class="section level4">
<h4>Loading data</h4>
<p>In order to run HybridFinder, three inputs must be provided to <code>HybridFinder</code></p>
<ol style="list-style-type: decimal">
<li>all de novo candidates export file - all denovo sequencing candidates, loaded into R as dataframe</li>
<li>DB search psm export file - database search results, loaded into R as dataframe.</li>
<li>folder path to the proteome database used.</li>
</ol>
<p>Please note that it is recommended to have a folder structure that looks as follows, as it helps keep all results organized:</p>
<ul>
<li>(parent folder)- Exp 1 (could be any name )
<ul>
<li>(child folder): first_run
<ul>
<li>denovo</li>
<li>db</li>
</ul></li>
<li>(2nd child folder): second_run</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>folder_Human_Liver_AUTD17 &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="st">&quot;./data/Human_Liver_AUTD17&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a>denovo_Human_Liver_AUTD17 &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;first_run&quot;</span>,<span class="st">&quot;all de novo candidates.csv&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">head=</span><span class="ot">TRUE</span>,<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb3-3"><a href="#cb3-3"></a>db_Human_Liver_AUTD17 &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;first_run&quot;</span>,<span class="st">&quot;DB search psm.csv&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">head=</span><span class="ot">TRUE</span>,<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb3-4"><a href="#cb3-4"></a>proteome_Human_Liver_AUTD17&lt;-<span class="st"> </span><span class="kw">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;uniprot-proteome-human_UP000005640-reviewed_validated.fasta&quot;</span>)</span></code></pre></div>
</div>
<div id="run-hybridfinder" class="section level4">
<h4>Run HybridFinder</h4>
<p>Once the inputs are loaded, running HybridFinder is a piece of cake. Please note that the HybridFinder function can use parallel computing in order to obtain results fast. It will be good to make sure whether the PC used can support that.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a></span>
<span id="cb4-2"><a href="#cb4-2"></a>results_HybridFinder_Human_Liver_AUTD17&lt;-<span class="st"> </span><span class="kw">HybridFinder</span>(<span class="dt">denovo_candidates =</span>  denovo_Human_Liver_AUTD17, <span class="dt">db_search =</span>  db_Human_Liver_AUTD17, <span class="dt">proteome_db =</span> proteome_Human_Liver_AUTD17, <span class="dt">with_parallel =</span> <span class="ot">FALSE</span>, <span class="dt">export_files =</span> <span class="ot">TRUE</span>, <span class="dt">export_dir=</span>folder_Human_Liver_AUTD17)</span></code></pre></div>
</div>
<div id="output" class="section level4">
<h4>Output</h4>
<p>The function returns a list composed of 3 elements</p>
<ul>
<li>a dataframe representing the HybridFinder output, containing the high confidence denovo peptides that made it through the different searches (listed above).</li>
<li>a vector containing all the hybrid candidate sequences</li>
<li>a list consisting of the merged reference+hybrid proteome.</li>
</ul>
<div id="ouput-1-hybridfinder-step1-output-hf_step1_output" class="section level5">
<h5>Ouput 1: HybridFinder Step1 output (HF_step1_output)</h5>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co">#display HybridFinder(HF) step1 output</span></span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw">print</span>(<span class="kw">head</span>(results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">1</span>]]))</span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co">#&gt;     Fraction     Scan      m/z    RT   Peptide Length Potential_spliceType ALC</span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="co">#&gt; 398        1 F1:12157 574.2689 61.13 NYGELFEKF      9                  cis  92</span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="co">#&gt; 10         1 F1:10576 569.8348 55.81 KLADRFLLY      9               Linear  89</span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="co">#&gt; 405        1 F1:12242 574.2689 61.13 DYGELFEKF      9                  cis  89</span></span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="co">#&gt; 97         1 F1:16670 575.7930 72.95 SYLEHLFEL      9               Linear  93</span></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">#&gt; 355        3 F3:16765 575.7935 72.81 SYLEHLFEL      9               Linear  91</span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">#&gt; 58         1 F1:14442 509.8185 68.48 LLDPHVVLL      9               Linear  82</span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co">#&gt;                                          proteome_database_used</span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">#&gt; 398 uniprot-proteome-human_UP000005640-reviewed_validated.fasta</span></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co">#&gt; 10  uniprot-proteome-human_UP000005640-reviewed_validated.fasta</span></span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="co">#&gt; 405 uniprot-proteome-human_UP000005640-reviewed_validated.fasta</span></span>
<span id="cb5-14"><a href="#cb5-14"></a><span class="co">#&gt; 97  uniprot-proteome-human_UP000005640-reviewed_validated.fasta</span></span>
<span id="cb5-15"><a href="#cb5-15"></a><span class="co">#&gt; 355 uniprot-proteome-human_UP000005640-reviewed_validated.fasta</span></span>
<span id="cb5-16"><a href="#cb5-16"></a><span class="co">#&gt; 58  uniprot-proteome-human_UP000005640-reviewed_validated.fasta</span></span></code></pre></div>
</div>
<div id="ouput-2-list-of-step1-candidate-hybrid-peptides" class="section level5">
<h5>Ouput 2: List of step1 candidate hybrid peptides</h5>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="co">#display list of candidate hybrid peptides</span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="kw">print</span>(<span class="kw">head</span>(results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">2</span>]]))</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co">#&gt; [1] &quot;KAVNLLLSY&quot; &quot;AKVNLLLSY&quot; &quot;KLADFRLLY&quot; &quot;KLADLFRLY&quot; &quot;NYGELFEKF&quot; &quot;DYGELFEKF&quot;</span></span></code></pre></div>
</div>
<div id="ouput-3-merged-proteome" class="section level5">
<h5>Ouput 3: merged proteome</h5>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co">#display the merged proteome</span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="kw">print</span>(<span class="kw">tail</span>(results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">3</span>]]))</span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&gt; $`sp|Q8TDM6-2|DLG5_HUMAN`</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt; [1] &quot;MEPQRRELLAQCQQSLAQAMTEVEAVLGLLEAAGALSPGERRQLDEEAGGAKAELLLKLLLAKERDHFQDLRAALEKTQPHLLPLLYLNGVVGPPQPAEGAGSTYSVLSTMPSDSESSSSLSSVGTTGKAPSPPPLLTDQQVNEKVENLSLQLRLMTRERNELRKRLAFATHGTAFDKRPYHRLNPDYERLKLQCVRAMSDLQSLQNQHTNALKRCEEVAKETDFYHTLHSRLLSDQTRLKDDVDMLRRENGQLLRERNLLQQSWEDMKRLHEEDQKELGDLRAQQQQVLKHNGSSELLNKLYDTAMDKLEVVKKDYDALRKRYSEKVALHNADLSRLEQLGEENQRLLKQTEMLTQQRDTALQLQHQCALSLRRFEALHHELNKATAQNKDLQWEMELLQSELTELRTTQVKTAKESEKYREERDAVYSEYKLLMSERDQVLSELDKLQTEVELAESKLKSSTSEKKAANEEMEALRQLKDTVTMDAGRANKEVELLRKQCKALCQELKEALQEADVAKCRRDWAFQERDKLVAERDSLRTLCDNLRRERDRAVSELAEALRSLDDTRKQKNDVSRELKELKEQMESQLEKEARFRQLMAHSSHDSALDTDSMEWETEVVEFERETEDLDLKALGFDMAEGVNEPCFPGDCGLFVTKVDKGSLADGRLRVNDWLLRLNDVDLLNKDKKQALKALLNGEGALNMVVRRRKSLGGKVVTPLHLNLSGQKDSGLSLENGVYAAAVLPGSPAAKEGSLAVGDRLVALNGLALDNKSLNECESLLRSCQDSLTLSLLKEQKCVPASGELSPELQEWAPYSPGHSSRHSNPPLYPSRPSVGTVPRSLTPSTTVSSLLRNPLYTVRSHRVGPCSSPPAARDAGPQGLHPSVQHQGRLSLDLSHRTCSDYSEMRATHGSNSLPSSARLGSSSNLQFKAERLKLPSTPRYPRSVVGSERGSVSHSECSTPPQSPLNLDTLSSCSQSQTSASTLPRLAVNPASLGERRKDRPYVEEPRHVKVQKGSEPLGLSLVSGEKGGLYVSKVTVGSLAHQAGLEYGDQLLEFNGLNLRSATEQQARLLLGQQCDTLTLLAQYNPHVHQLSSHSRSSSHLDPAGTHSTLQGSGTTTPEHPSVLDPLMEQDEGPSTPPAKQSSSRLAGDANKKTLEPRVVFLKKSQLELGVHLCGGNLHGVFVAEVEDDSPAKGPDGLVPGDLLLEYGSLDVRNKTVEEVYVEMLKPRDGVRLKVQYRPEEFTKAKGLPGDSFYLRALYDRLADVEQELSFKKDDLLYVDDTLPQGTFGSWMAWQLDENAQKLQRGQLPSKYVMDQEFSRRLSMSEVKDDNSATKTLSAAARRSFFRRKHKHKRSGSKDGKDLLALDAFSSDSLPLFEDSVSLAYQRVQKVDCTALRPVLLLGPLLDVVKEMLVNEAPGKFCRCPLEVMKASQQALERGVKDCLFVDYKRRSGHFDVTTVASLKELTEKNRHCLLDLAPHALERLHHMHLYPLVLFLHYKSAKHLKEQRDPLYLRDKVTQRHSKEQFEAAQKLEQEYSRYFTGVLQGGALSSLCTQLLAMVNQEQNKVLWLPACPL&quot;</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">#&gt; attr(,&quot;name&quot;)</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt; [1] &quot;sp|Q8TDM6-2|DLG5_HUMAN&quot;</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt; attr(,&quot;Annot&quot;)</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt; [1] &quot;&gt;sp|Q8TDM6-2|DLG5_HUMAN Isoform 2 of Disks large homolog 5 OS=Homo sapiens OX=9606 GN=DLG5&quot;</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt; attr(,&quot;class&quot;)</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#&gt; [1] &quot;SeqFastaAA&quot;</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="co">#&gt; </span></span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="co">#&gt; $`sp|Q8TDM6-3|DLG5_HUMAN`</span></span>
<span id="cb7-13"><a href="#cb7-13"></a><span class="co">#&gt; [1] &quot;MEPQRRELLAQCQQSLAQAMTEVEAVLGLLEAAGALSPGERRQLDEEAGGAKAELLLKLLLAKERDHFQDLRAALEKTQPHLLPLLYLNGVVGPPQPAEGAGSTYSVLSTMPSDSESSSSLSSVGTTGKELKEQMESQLEKEARFRQLMAHSSHDSALDTDSMEWETEVVEFERETEDLDLKALGFDMAEGVNEPCFPGDCGLFVTKVDKGSLADGRLRVNDWLLRLNDVDLLNKDKKQALKALLNGEGALNMVVRRRKSLGGKVVTPLHLNLSGQKDSGLSLENGVYAAAVLPGSPAAKEGSLAVGDRLVALNGLALDNKSLNECESLLRSCQDSLTLSLLKVFPQSSSWSGQNLFENLKDSDKMLSFRAHGPEVQAHNKRNLLQHNNSTQTDLFYTDRLEDRKEPGPPGGSSSFLHKPFPGGPLQVCPQACPSASERSLSSFRSDASGDRGFGLVDVRGRRPLLPFETEVGPCGVGEASLDKADSEGSNSGGTWPKAMLSSTAVPEKLSVYKKPKQRKSLFDPNTFKRPQTPPKLDYLLPGPGPAHSPQPSKRAGPLTPPKPPRRSDSLKFQHRLETSSESEATLVGSSPSTSPPSALPPDVDPGEPMHASPPRKARVRLASSYYPEGDGDSSHLPAKKSCDEDLTSQKVDELGQKRRRPKSAPSFRPKLAPVVLPAQFLEV&quot;</span></span>
<span id="cb7-14"><a href="#cb7-14"></a><span class="co">#&gt; attr(,&quot;name&quot;)</span></span>
<span id="cb7-15"><a href="#cb7-15"></a><span class="co">#&gt; [1] &quot;sp|Q8TDM6-3|DLG5_HUMAN&quot;</span></span>
<span id="cb7-16"><a href="#cb7-16"></a><span class="co">#&gt; attr(,&quot;Annot&quot;)</span></span>
<span id="cb7-17"><a href="#cb7-17"></a><span class="co">#&gt; [1] &quot;&gt;sp|Q8TDM6-3|DLG5_HUMAN Isoform 3 of Disks large homolog 5 OS=Homo sapiens OX=9606 GN=DLG5&quot;</span></span>
<span id="cb7-18"><a href="#cb7-18"></a><span class="co">#&gt; attr(,&quot;class&quot;)</span></span>
<span id="cb7-19"><a href="#cb7-19"></a><span class="co">#&gt; [1] &quot;SeqFastaAA&quot;</span></span>
<span id="cb7-20"><a href="#cb7-20"></a><span class="co">#&gt; </span></span>
<span id="cb7-21"><a href="#cb7-21"></a><span class="co">#&gt; $`sp|Q8TDM6-4|DLG5_HUMAN`</span></span>
<span id="cb7-22"><a href="#cb7-22"></a><span class="co">#&gt; [1] &quot;MPSDSESSSSLSSVGTTGKAPSPPPLLTDQQVNEKVENLSLQLRLMTRERNELRKRLAFATHGTAFDKRPYHRLNPDYERLKLQCVRAMSDLQSLQNQHTNALKRCEEVAKETDFYHTLHSRLLSDQTRLKDDVDMLRRENGQLLRERNLLQQSWEDMKRLHEEDQKELGDLRAQQQQVLKHNGSSELLNKLYDTAMDKLEVVKKDYDALRKRYSEKVALHNADLSRLEQLGEENQRLLKQTEMLTQQRDTALQLQHQCALSLRRFEALHHELNKATAQNKDLQWEMELLQSELTELRTTQVKTAKESEKYREERDAVYSEYKLLMSERDQVLSELDKLQTEVELAESKLKSSTSEKKAANEEMEALRQLKDTVTMDAGRANKEVELLRKQCKALCQELKEALQEADVAKCRRDWAFQERDKLVAERDSLRTLCDNLRRERDRAVSELAEALRSLDDTRKQKNDVSRELKELKEQMESQLEKEARFRQLMAHSSHDSALDTDSMEWETEVVEFERETEDLDLKALGFDMAEGVNEPCFPGDCGLFVTKVDKGSLADGRLRVNDWLLRLNDVDLLNKDKKQALKALLNGEGALNMVVRRRKSLGGKVVTPLHLNLSGQKDSGLSLENGVYAAAVLPGSPAAKEGSLAVGDRLVALNGLALDNKSLNECESLLRSCQDSLTLSLLKVFPQSSSWSGQNLFENLKDSDKMLSFRAHGPEVQAHNKRNLLQHNNSTQTDLFYTDRLEDRKEPGPPGGSSSFLHKPFPGGPLQVCPQACPSASERSLSSFRSDASGDRGFGLVDVRGRRPLLPFETEVGPCGVGEASLDKADSEGSNSGGTWPKAMLSSTAVPEKLSVYKKPKQRKSLFDPNTFKRPQTPPKLDYLLPGPGPAHSPQPSKRAGPLTPPKPPRRSDSLKFQHRLETSSESEATLVGSSPSTSPPSALPPDVDPGEPMHASPPRKARVRLASSYYPEGDGDSSHLPAKKSCDEDLTSQKVDELGQKRRRPKSAPSFRPKLAPVVLPAQFLEEQKCVPASGELSPELQEWAPYSPGHSSRHSNPPLYPSRPSVGTVPRSLTPSTTVSSLLRNPLYTVRSHRVGPCSSPPAARDAGPQGLHPSVQHQGRLSLDLSHRTCSDYSEMRATHGSNSLPSSARLGSSSNLQFKAERLKLPSTPRYPRSVVGSERGSVSHSECSTPPQSPLNLDTLSSCSQSQTSASTLPRLAVNPASLGERRKDRPYVEEPRHVKVQKGSEPLGLSLVSGEKGGLYVSKVTVGSLAHQAGLEYGDQLLEFNGLNLRSATEQQARLLLGQQCDTLTLLAQYNPHVHQLSSHSRSSSHLDPAGTHSTLQGSGTTTPEHPSVLDPLMEQDEGPSTPPAKQSSSRLAGDANKKTLEPRVVFLKKSQLELGVHLCGGNLHGVFVAEVEDDSPAKGPDGLVPGDLLLEYGSLDVRNKTVEEVYVEMLKPRDGVRLKVQYRPEEFTKAKGLPGDSFYLRALYDRLADVEQELSFKKDDLLYVDDTLPQGTFGSWMAWQLDENAQKLQRGQLPSKYVMDQEFSRRLSMSEVKDDNSATKTLSAAARRSFFRRKHKHKRSGSKDGKDLLALDAFSSDSLPLFEDSVSLAYQRVQKVDCTALRPVLLLGPLLDVVKEMLVNEAPGKFCRCPLEVMKASQQALERGVKDCLFVDYKRRSGHFDVTTVASLKELTEKNRHCLLDLAPHALERLHHMHLYPLVLFLHYKSAKHLKEQRDPLYLRDKVTQRHSKEQFEAAQKLEQEYSRYFTGVLQGGALSSLCTQLLAMVNQEQNKVLWLPACPL&quot;</span></span>
<span id="cb7-23"><a href="#cb7-23"></a><span class="co">#&gt; attr(,&quot;name&quot;)</span></span>
<span id="cb7-24"><a href="#cb7-24"></a><span class="co">#&gt; [1] &quot;sp|Q8TDM6-4|DLG5_HUMAN&quot;</span></span>
<span id="cb7-25"><a href="#cb7-25"></a><span class="co">#&gt; attr(,&quot;Annot&quot;)</span></span>
<span id="cb7-26"><a href="#cb7-26"></a><span class="co">#&gt; [1] &quot;&gt;sp|Q8TDM6-4|DLG5_HUMAN Isoform 4 of Disks large homolog 5 OS=Homo sapiens OX=9606 GN=DLG5&quot;</span></span>
<span id="cb7-27"><a href="#cb7-27"></a><span class="co">#&gt; attr(,&quot;class&quot;)</span></span>
<span id="cb7-28"><a href="#cb7-28"></a><span class="co">#&gt; [1] &quot;SeqFastaAA&quot;</span></span>
<span id="cb7-29"><a href="#cb7-29"></a><span class="co">#&gt; </span></span>
<span id="cb7-30"><a href="#cb7-30"></a><span class="co">#&gt; $`sp|Q8TDM6-5|DLG5_HUMAN`</span></span>
<span id="cb7-31"><a href="#cb7-31"></a><span class="co">#&gt; [1] &quot;MRATHGSNSLPSSARLGSSSNLQFKAERLKLPSTPRYPRSVVGSERGSVSHSECSTPPQSPLNLDTLSSCSQSQTSASTLPRLAVNPASLGERRKDRPYVEEPRHVKVQKGSEPLGLSLVSGEKGGLYVSKVTVGSLAHQAGLEYGDQLLEFNGLNLRSATEQQARLLLGQQCDTLTLLAQYNPHVHQLSSHSRSSSHLDPAGTHSTLQGSGTTTPEHPSVLDPLMEQDEGPSTPPAKQSSSRLAGDANKKTLEPRVVFLKKSQLELGVHLCGGNLHGVFVAEVEDDSPAKGPDGLVPGDLLLEYGSLDVRNKTVEEVYVEMLKPRDGVRLKVQYRPEEFTKAKGLPGDSFYLRALYDRLADVEQELSFKKDDLLYVDDTLPQGTFGSWMAWQLDENAQKLQRGQLPSKYVMDQEFSRRLSMSEVKDDNSATKTLSAAARRSFFRRKHKHKRSGSKDGKDLLALDAFSSDSLPLFEDSVSLAYQRVQKVDCTALRPVLLLGPLLDVVKEMLVNEAPGKFCRCPLEVMKASQQALERGVKDCLFVDYKRRSGHFDVTTVASLKELTEKNRHCLLDLAPHALERLHHMHLYPLVLFLHYKSAKHLKEQRDPLYLRDKVTQRHSKEQFEAAQKLEQEYSRYFTGVLQGGALSSLCTQLLAMVNQEQNKVLWLPACPL&quot;</span></span>
<span id="cb7-32"><a href="#cb7-32"></a><span class="co">#&gt; attr(,&quot;name&quot;)</span></span>
<span id="cb7-33"><a href="#cb7-33"></a><span class="co">#&gt; [1] &quot;sp|Q8TDM6-5|DLG5_HUMAN&quot;</span></span>
<span id="cb7-34"><a href="#cb7-34"></a><span class="co">#&gt; attr(,&quot;Annot&quot;)</span></span>
<span id="cb7-35"><a href="#cb7-35"></a><span class="co">#&gt; [1] &quot;&gt;sp|Q8TDM6-5|DLG5_HUMAN Isoform 5 of Disks large homolog 5 OS=Homo sapiens OX=9606 GN=DLG5&quot;</span></span>
<span id="cb7-36"><a href="#cb7-36"></a><span class="co">#&gt; attr(,&quot;class&quot;)</span></span>
<span id="cb7-37"><a href="#cb7-37"></a><span class="co">#&gt; [1] &quot;SeqFastaAA&quot;</span></span>
<span id="cb7-38"><a href="#cb7-38"></a><span class="co">#&gt; </span></span>
<span id="cb7-39"><a href="#cb7-39"></a><span class="co">#&gt; $`sp|denovo_HF_fake_protein1`</span></span>
<span id="cb7-40"><a href="#cb7-40"></a><span class="co">#&gt; [1] &quot;KAVNLLLSYAKVNLLLSYKLADFRLLYKLADLFRLYNYGELFEKFDYGELFEKFDYGELFQKFKLADFLRLYYKPSPFFVFYNLPWLENLLEPFLLPTLPELFLLPTLLYEQFVPLLLEYQFVPLLPLEFLLPTLLTTSWMSLKEQLPLRLSAQELPLRLSASEAPPTNGAALPYFSPCLELLPENLLHALMEDLLKLLAMENLLKLLAYPDLNFRNLLPVDLQRYLLPVNLQRYLLYEVLLKNFFPYYAPELLPFYYAPELLFSVHMVTHFLLYYASNYRRFLVGSLPKESAPPTNQAFENGEWRELQLADLFRLYEFTQHLFEL&quot;</span></span>
<span id="cb7-41"><a href="#cb7-41"></a><span class="co">#&gt; </span></span>
<span id="cb7-42"><a href="#cb7-42"></a><span class="co">#&gt; $`sp|denovo_HF_fake_protein2`</span></span>
<span id="cb7-43"><a href="#cb7-43"></a><span class="co">#&gt; [1] &quot;LTMNLVQELLTMDLVQELLTMDLVQQLLWDLSLTRLYLNPSFTVLYKPSFPFVFKYPSMLFVFYKPSLMFVFKYPSLMFVFDQDLRSMATALLYYASNRYLLYYASRNYSLVMTQTPKFLLTTMSLGSFLLTERYGSFFYGKQAVQFYKVYTSVSWMMALLTHGLLMMAKVFVFSYLPLAHFMAYVSELFPAFSFVSELFPAFSYLHELFELNYLPWLELNYVLAQAVLSLTVVMTQTPKFRYFSTSVSWYRFSTSVSWLTFVPGAMVLPVDLTAFQLLPVNLATFQLLPVDLATFQLLPVNLKSLTMSYLEHLFYTLPEMWFPLL&quot;</span></span></code></pre></div>
</div>
</div>
<div id="export" class="section level4">
<h4>Export</h4>
<p>If export is set to <code>TRUE</code> and a valid directory is provided in <code>export_dir</code>, then the results are exported .csv, .csv and .fasta format, respectively.</p>
<p>Even if the export parameters were not set at the beginning, the results returned can always be exported with the <code>export_HybridFinder_results</code> function as long as as the results obtained from the HybridFinder function are stored which is also indicated in the results_list parameter of the <code>export_HybridFinder_results</code> function.</p>
</div>
</div>
</div>
<div id="interim-external-step-second-database-search-using-the-merged-proteome" class="section level2">
<h2>Interim external step: Second database search using the merged proteome</h2>
<p>After finishing this, a second database search has to be done on the raw MS however with the merged proteome (.fasta) exported from the HybridFinder function results.</p>
</div>
<div id="step2" class="section level2">
<h2>Step2</h2>
<p>The second step in RHybridFinder consists of either using <code>checknetMHCpan</code> or <code>step2_wo_netMHCpan</code>, while using the results from step 1 in order to retrieve for the final list of peptides which includes the hybrid candidates, their potential splice types.</p>
<div id="checknetmhcpan" class="section level3">
<h3>checknetMHCpan</h3>
<div id="description-1" class="section level4">
<h4>Description</h4>
<p>the <code>checknetMHCpan</code> function represents step 2 of Faridi et al. (2018)’s workflow and also features the use of netMHCpan (Jurtz et al., 2017, Reynisson et al., 2020) for obtaining the peptide-MHC-I predicted binding affinities. Please note that netMHCpan needs to be installed in order to be able to run this function. The package also contains a function that runs step2 without netMHCpan (Please refer to the <code>step2_wo_netMHCpan</code> part).</p>
</div>
<div id="loading-data-1" class="section level4">
<h4>Loading data</h4>
<p>In order to run checknetMHCpan, four inputs must be provided to <code>checknetMHCpan</code></p>
<ol style="list-style-type: decimal">
<li>netmhcpan_directory: the directory in which netMHCpan is located in (i.e ‘/usr/bin/’ or ‘/usr/bin/local/’)</li>
<li>netmhcpan_alleles: the alleles to be tested again, in a vector format if multiple (i.e alleles&lt;- c(‘HLA-A<em>03:01’, ’HLA-A</em>24:02’))</li>
<li>peptide_rerun: the database search results from the 2nd run loaded into R as a dataframe</li>
<li>HF_step1_output: the dataframe of the first element of the <code>HybridFinder</code> output.</li>
</ol>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>netmhcpan_dir&lt;-<span class="st"> &#39;/usr/bin/&#39;</span></span>
<span id="cb8-2"><a href="#cb8-2"></a></span>
<span id="cb8-3"><a href="#cb8-3"></a>alleles_Human_liver_AUTD17&lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;HLA-A*03:01&quot;</span>, <span class="st">&quot;HLA-A*24:02&quot;</span>, <span class="st">&quot;HLA-B*35:03&quot;</span>, <span class="st">&quot;HLA-B*45:01&quot;</span>, <span class="st">&quot;HLA-C*04:01&quot;</span>, <span class="st">&quot;HLA-C*16:01&quot;</span>)</span>
<span id="cb8-4"><a href="#cb8-4"></a></span>
<span id="cb8-5"><a href="#cb8-5"></a>db_rerun_Human_liver_AUTD17 &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;second_run&quot;</span>,<span class="st">&quot;DB search psm.csv&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">head=</span><span class="ot">TRUE</span>,<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb8-6"><a href="#cb8-6"></a></span>
<span id="cb8-7"><a href="#cb8-7"></a>HF_output_Human_liver_AUTD17&lt;-<span class="st"> </span>results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">1</span>]]</span></code></pre></div>
</div>
<div id="run-checknetmhcpan" class="section level4">
<h4>Run checknetMHCpan</h4>
<p>Once the inputs are loaded, running checknetMHCpan is easier than ABC.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a></span>
<span id="cb9-2"><a href="#cb9-2"></a>results_checknetMHCpan_Human_Liver_AUTD17&lt;-<span class="st"> </span><span class="kw">checknetMHCpan</span>(<span class="dt">netmhcpan_directory =</span> netmhcpan_dir, <span class="dt">netmhcpan_alleles =</span> alleles_Human_liver_AUTD17, <span class="dt">peptide_rerun =</span> db_rerun_Human_liver_AUTD17, <span class="dt">HF_step1_output =</span> HF_output_Human_liver_AUTD17, <span class="dt">export_files =</span> <span class="ot">TRUE</span>, <span class="dt">export_dir=</span>folder_Human_Liver_AUTD17)</span></code></pre></div>
</div>
<div id="output-1" class="section level4">
<h4>Output</h4>
<p>The function returns a list composed of 3 elements: - the netMHCpan results in long format, that is the binding affinity results are displayed for each peptide with a given allele from those chosen. - the netMHCpan results in wide format, that is the binding affinity levels per peptide summarized for all HLA alleles chosen. - the database results with the respective potential splice types retrieved from step 1</p>
<div id="ouput-1-netmhcpan-results-in-long-format" class="section level5">
<h5>Ouput 1: netMHCpan results in long format</h5>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co">#display netmhcpan output(long version)</span></span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="kw">print</span>(<span class="kw">head</span>(results_checknetMHCpan_Human_Liver_AUTD17[[<span class="dv">1</span>]]))</span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="co">#&gt;    Pos         HLA    Peptide      Core Of Gp Gl Ip Il      Icore Identity</span></span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="co">#&gt; 5    1 HLA-A*03:01  DYENLFLKF DYENLFLKF  0  0  0  0  0  DYENLFLKF  PEPLIST</span></span>
<span id="cb10-5"><a href="#cb10-5"></a><span class="co">#&gt; 6    1 HLA-A*03:01  RYFSTSVSW RYFSTSVSW  0  0  0  0  0  RYFSTSVSW  PEPLIST</span></span>
<span id="cb10-6"><a href="#cb10-6"></a><span class="co">#&gt; 7    1 HLA-A*03:01 AFSHLLLTTM AFSHLLLTM  0  7  1  0  0 AFSHLLLTTM  PEPLIST</span></span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="co">#&gt; 8    1 HLA-A*03:01  PYMARVAFF PYMARVAFF  0  0  0  0  0  PYMARVAFF  PEPLIST</span></span>
<span id="cb10-8"><a href="#cb10-8"></a><span class="co">#&gt; 9    1 HLA-A*03:01  LYRPTAAAF LYRPTAAAF  0  0  0  0  0  LYRPTAAAF  PEPLIST</span></span>
<span id="cb10-9"><a href="#cb10-9"></a><span class="co">#&gt; 10   1 HLA-A*03:01 FPVELAKYYM FVELAKYYM  0  1  1  0  0 FPVELAKYYM  PEPLIST</span></span>
<span id="cb10-10"><a href="#cb10-10"></a><span class="co">#&gt;        Score Aff(nM)   %Rank  BindLevel strongBinder weakBinder  noneBinder</span></span>
<span id="cb10-11"><a href="#cb10-11"></a><span class="co">#&gt; 5  0.0232400 38883.6 68.5113 Non binder                         HLA-A*03:01</span></span>
<span id="cb10-12"><a href="#cb10-12"></a><span class="co">#&gt; 6  0.0939940 18084.0 14.1393 Non binder                         HLA-A*03:01</span></span>
<span id="cb10-13"><a href="#cb10-13"></a><span class="co">#&gt; 7  0.0874160 19418.0 15.5482 Non binder                         HLA-A*03:01</span></span>
<span id="cb10-14"><a href="#cb10-14"></a><span class="co">#&gt; 8  0.0423440 31622.6 38.9120 Non binder                         HLA-A*03:01</span></span>
<span id="cb10-15"><a href="#cb10-15"></a><span class="co">#&gt; 9  0.0728480 22733.1 19.7710 Non binder                         HLA-A*03:01</span></span>
<span id="cb10-16"><a href="#cb10-16"></a><span class="co">#&gt; 10 0.0459210 30422.0 35.4191 Non binder                         HLA-A*03:01</span></span>
<span id="cb10-17"><a href="#cb10-17"></a><span class="co">#&gt;    Potential_spliceType</span></span>
<span id="cb10-18"><a href="#cb10-18"></a><span class="co">#&gt; 5                Linear</span></span>
<span id="cb10-19"><a href="#cb10-19"></a><span class="co">#&gt; 6                 trans</span></span>
<span id="cb10-20"><a href="#cb10-20"></a><span class="co">#&gt; 7                Linear</span></span>
<span id="cb10-21"><a href="#cb10-21"></a><span class="co">#&gt; 8                Linear</span></span>
<span id="cb10-22"><a href="#cb10-22"></a><span class="co">#&gt; 9                Linear</span></span>
<span id="cb10-23"><a href="#cb10-23"></a><span class="co">#&gt; 10               Linear</span></span></code></pre></div>
</div>
<div id="ouput-2-netmhcpan-results-in-wide-format" class="section level5">
<h5>Ouput 2: netMHCpan results in wide format</h5>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co">#display netmhcpan output tidied version (wide)</span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw">print</span>(<span class="kw">head</span>(results_checknetMHCpan_Human_Liver_AUTD17[[<span class="dv">2</span>]]))</span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="co">#&gt;        Peptide                        strongBinder              weakBinder</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="co">#&gt; 1  AAEYPSVTNYL                                                 HLA-A*24:02</span></span>
<span id="cb11-5"><a href="#cb11-5"></a><span class="co">#&gt; 7    AAFFEEPEL                                     HLA-B*35:03,HLA-C*16:01</span></span>
<span id="cb11-6"><a href="#cb11-6"></a><span class="co">#&gt; 13  AAMLDTVVFK                         HLA-A*03:01                        </span></span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co">#&gt; 19   AANPHSFVF HLA-B*35:03,HLA-C*04:01,HLA-C*16:01             HLA-A*24:02</span></span>
<span id="cb11-8"><a href="#cb11-8"></a><span class="co">#&gt; 25   AANPNGRYY                                                 HLA-C*16:01</span></span>
<span id="cb11-9"><a href="#cb11-9"></a><span class="co">#&gt; 31  AAPPQLRALL                                                            </span></span>
<span id="cb11-10"><a href="#cb11-10"></a><span class="co">#&gt;                                                                 noneBinder</span></span>
<span id="cb11-11"><a href="#cb11-11"></a><span class="co">#&gt; 1              HLA-A*03:01,HLA-B*35:03,HLA-B*45:01,HLA-C*04:01,HLA-C*16:01</span></span>
<span id="cb11-12"><a href="#cb11-12"></a><span class="co">#&gt; 7                          HLA-A*03:01,HLA-A*24:02,HLA-B*45:01,HLA-C*04:01</span></span>
<span id="cb11-13"><a href="#cb11-13"></a><span class="co">#&gt; 13             HLA-A*24:02,HLA-B*35:03,HLA-B*45:01,HLA-C*04:01,HLA-C*16:01</span></span>
<span id="cb11-14"><a href="#cb11-14"></a><span class="co">#&gt; 19                                                 HLA-A*03:01,HLA-B*45:01</span></span>
<span id="cb11-15"><a href="#cb11-15"></a><span class="co">#&gt; 25             HLA-A*03:01,HLA-A*24:02,HLA-B*35:03,HLA-B*45:01,HLA-C*04:01</span></span>
<span id="cb11-16"><a href="#cb11-16"></a><span class="co">#&gt; 31 HLA-A*03:01,HLA-A*24:02,HLA-B*35:03,HLA-B*45:01,HLA-C*04:01,HLA-C*16:01</span></span>
<span id="cb11-17"><a href="#cb11-17"></a><span class="co">#&gt;    %Rank.HLA-B*35:03 %Rank.HLA-A*24:02 %Rank.HLA-C*04:01 %Rank.HLA-A*03:01</span></span>
<span id="cb11-18"><a href="#cb11-18"></a><span class="co">#&gt; 1            13.2675            1.3806            2.8884           19.5261</span></span>
<span id="cb11-19"><a href="#cb11-19"></a><span class="co">#&gt; 7             1.0785           27.1675           10.0025           43.1042</span></span>
<span id="cb11-20"><a href="#cb11-20"></a><span class="co">#&gt; 13            8.8987           18.3151           11.6722            0.1091</span></span>
<span id="cb11-21"><a href="#cb11-21"></a><span class="co">#&gt; 19            0.3034            1.2434            0.2483            8.6520</span></span>
<span id="cb11-22"><a href="#cb11-22"></a><span class="co">#&gt; 25            7.0735           38.2717            3.5019            5.3500</span></span>
<span id="cb11-23"><a href="#cb11-23"></a><span class="co">#&gt; 31            6.9323           10.8503            5.7910           44.3338</span></span>
<span id="cb11-24"><a href="#cb11-24"></a><span class="co">#&gt;    %Rank.HLA-B*45:01 %Rank.HLA-C*16:01 strongBinder_count weakBinder_count</span></span>
<span id="cb11-25"><a href="#cb11-25"></a><span class="co">#&gt; 1             5.5163           10.6324                  0                1</span></span>
<span id="cb11-26"><a href="#cb11-26"></a><span class="co">#&gt; 7            45.5616            0.6543                  0                2</span></span>
<span id="cb11-27"><a href="#cb11-27"></a><span class="co">#&gt; 13           10.4367            7.4449                  1                0</span></span>
<span id="cb11-28"><a href="#cb11-28"></a><span class="co">#&gt; 19            9.1652            0.0156                  3                1</span></span>
<span id="cb11-29"><a href="#cb11-29"></a><span class="co">#&gt; 25           21.7060            0.5745                  0                1</span></span>
<span id="cb11-30"><a href="#cb11-30"></a><span class="co">#&gt; 31           73.9918            4.4549                  0                0</span></span>
<span id="cb11-31"><a href="#cb11-31"></a><span class="co">#&gt;    noneBinder_count Potential_spliceType</span></span>
<span id="cb11-32"><a href="#cb11-32"></a><span class="co">#&gt; 1                 5               Linear</span></span>
<span id="cb11-33"><a href="#cb11-33"></a><span class="co">#&gt; 7                 4               Linear</span></span>
<span id="cb11-34"><a href="#cb11-34"></a><span class="co">#&gt; 13                5               Linear</span></span>
<span id="cb11-35"><a href="#cb11-35"></a><span class="co">#&gt; 19                2               Linear</span></span>
<span id="cb11-36"><a href="#cb11-36"></a><span class="co">#&gt; 25                5               Linear</span></span>
<span id="cb11-37"><a href="#cb11-37"></a><span class="co">#&gt; 31                6               Linear</span></span></code></pre></div>
</div>
<div id="ouput-3-database-search-results-updated" class="section level5">
<h5>Ouput 3: Database search results updated</h5>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co">#display the updated database search results with the categorizations from step1</span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="kw">print</span>(<span class="kw">head</span>(results_checknetMHCpan_Human_Liver_AUTD17[[<span class="dv">3</span>]]))</span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="co">#&gt;     Peptide X.10lgP     Mass Length ppm      m.z Z   RT   Area Fraction    Id</span></span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="co">#&gt; 1 DYENLFLKF   23.98 1187.586      9 0.1 594.8004 2 73.2 243590        3 44426</span></span>
<span id="cb12-5"><a href="#cb12-5"></a><span class="co">#&gt; 2 DYENLFLKF   23.72 1187.586      9 0.1 594.8004 2 73.2 243590        3 44427</span></span>
<span id="cb12-6"><a href="#cb12-6"></a><span class="co">#&gt; 3 DYENLFLKF   23.13 1187.586      9 0.1 594.8004 2 73.2 243590        3 44428</span></span>
<span id="cb12-7"><a href="#cb12-7"></a><span class="co">#&gt; 4 DYENLFLKF   22.98 1187.586      9 0.1 594.8004 2 73.2 243590        3 44429</span></span>
<span id="cb12-8"><a href="#cb12-8"></a><span class="co">#&gt; 5 DYENLFLKF   22.70 1187.586      9 0.1 594.8004 2 73.2 243590        3 44430</span></span>
<span id="cb12-9"><a href="#cb12-9"></a><span class="co">#&gt; 6 DYENLFLKF   20.88 1187.586      9 0.1 594.8004 2 73.2 243590        3 44432</span></span>
<span id="cb12-10"><a href="#cb12-10"></a><span class="co">#&gt;       Scan from.Chimera</span></span>
<span id="cb12-11"><a href="#cb12-11"></a><span class="co">#&gt; 1 F3:16451           No</span></span>
<span id="cb12-12"><a href="#cb12-12"></a><span class="co">#&gt; 2 F3:16501           No</span></span>
<span id="cb12-13"><a href="#cb12-13"></a><span class="co">#&gt; 3 F3:16546           No</span></span>
<span id="cb12-14"><a href="#cb12-14"></a><span class="co">#&gt; 4 F3:16609           No</span></span>
<span id="cb12-15"><a href="#cb12-15"></a><span class="co">#&gt; 5 F3:16633           No</span></span>
<span id="cb12-16"><a href="#cb12-16"></a><span class="co">#&gt; 6 F3:16686           No</span></span>
<span id="cb12-17"><a href="#cb12-17"></a><span class="co">#&gt;                                                 Source.File</span></span>
<span id="cb12-18"><a href="#cb12-18"></a><span class="co">#&gt; 1 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb12-19"><a href="#cb12-19"></a><span class="co">#&gt; 2 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb12-20"><a href="#cb12-20"></a><span class="co">#&gt; 3 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb12-21"><a href="#cb12-21"></a><span class="co">#&gt; 4 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb12-22"><a href="#cb12-22"></a><span class="co">#&gt; 5 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb12-23"><a href="#cb12-23"></a><span class="co">#&gt; 6 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb12-24"><a href="#cb12-24"></a><span class="co">#&gt;                  Accession PTM AScore Found.By Peptide_no_mods</span></span>
<span id="cb12-25"><a href="#cb12-25"></a><span class="co">#&gt; 1 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb12-26"><a href="#cb12-26"></a><span class="co">#&gt; 2 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb12-27"><a href="#cb12-27"></a><span class="co">#&gt; 3 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb12-28"><a href="#cb12-28"></a><span class="co">#&gt; 4 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb12-29"><a href="#cb12-29"></a><span class="co">#&gt; 5 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb12-30"><a href="#cb12-30"></a><span class="co">#&gt; 6 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb12-31"><a href="#cb12-31"></a><span class="co">#&gt;   Potential_spliceType</span></span>
<span id="cb12-32"><a href="#cb12-32"></a><span class="co">#&gt; 1               Linear</span></span>
<span id="cb12-33"><a href="#cb12-33"></a><span class="co">#&gt; 2               Linear</span></span>
<span id="cb12-34"><a href="#cb12-34"></a><span class="co">#&gt; 3               Linear</span></span>
<span id="cb12-35"><a href="#cb12-35"></a><span class="co">#&gt; 4               Linear</span></span>
<span id="cb12-36"><a href="#cb12-36"></a><span class="co">#&gt; 5               Linear</span></span>
<span id="cb12-37"><a href="#cb12-37"></a><span class="co">#&gt; 6               Linear</span></span></code></pre></div>
</div>
</div>
<div id="export-1" class="section level4">
<h4>Export</h4>
<p>If export is set to <code>TRUE</code> and a valid directory is provided in <code>export_dir</code>, then the results are exported .csv, .tsv (tab-separated) and .csv format, respectively.</p>
<p>Even if the export parameters were not set at the beginning, the results returned can always be exported with the <code>export_checknetMHCpan_results</code> function as long as as the results obtained from the checknetMHCpan function are stored which is also indicated in the results_list parameter of the <code>export_checknetMHCpan_results</code> function.</p>
</div>
</div>
<div id="step2_wo_netmhcpan" class="section level3">
<h3>step2_wo_netMHCpan</h3>
<p>The <code>step2_wo_netMHCpan</code>, removes peptide modifications and prepare a peptide (.pep) file for use in webversion of netMHCpan, in case netMHCpan is not installed, OS is windows or the user would like to run in another software. Additionally, the function matches peptide sequences in the database search rerun (the second database search where the merged proteome was used), with the predicted splice type obtained from step 1.</p>
<div id="description-2" class="section level4">
<h4>Description</h4>
<p>The step2_wo_netMHCpan, removes peptide modifications and runs netMHCpan on peptides between 9 and 12-mers. Additionally, the function matches peptide sequences in the database search rerun (the second database search where the merged proteome was used), with predicted splice type obtained from step 1.</p>
</div>
<div id="loading-data-2" class="section level4">
<h4>Loading data</h4>
<p>In order to run checknetMHCpan, four inputs must be provided to <code>checknetMHCpan</code></p>
<ol style="list-style-type: decimal">
<li>peptide_rerun: the database search results from the 2nd run loaded into R as a dataframe</li>
<li>HF_step1_output: the dataframe of the first element of the <code>HybridFinder</code> output.</li>
</ol>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>db_rerun_Human_liver_AUTD17 &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">file.path</span>(folder_Human_Liver_AUTD17, <span class="st">&quot;second_run&quot;</span>,<span class="st">&quot;DB search psm.csv&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>, <span class="dt">head=</span><span class="ot">TRUE</span>,<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb13-2"><a href="#cb13-2"></a></span>
<span id="cb13-3"><a href="#cb13-3"></a>HF_output_Human_liver_AUTD17&lt;-<span class="st"> </span>results_HybridFinder_Human_Liver_AUTD17[[<span class="dv">1</span>]]</span></code></pre></div>
</div>
<div id="run-step2_wo_netmhcpan" class="section level4">
<h4>Run step2_wo_netMHCpan</h4>
<p>Once the inputs are loaded, running step2_wo_netMHCpan is easier than ABC.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a></span>
<span id="cb14-2"><a href="#cb14-2"></a>results_step2_Human_Liver_AUTD17&lt;-<span class="st"> </span><span class="kw">step2_wo_netMHCpan</span>(<span class="dt">peptide_rerun =</span> db_rerun_Human_liver_AUTD17, <span class="dt">HF_step1_output =</span> HF_output_Human_liver_AUTD17, <span class="dt">export_files =</span> <span class="ot">TRUE</span>, <span class="dt">export_dir=</span>folder_Human_Liver_AUTD17)</span></code></pre></div>
</div>
<div id="output-2" class="section level4">
<h4>Output</h4>
<p>The function returns a list composed of 2 elements: - a character vector containing the list of unique peptides from the database search rerun without modifications and of length 9 to 12 amino acids - the database results with the respective potential splice types retrieved from step 1</p>
<div id="ouput-1-netmhcpan-ready-input" class="section level5">
<h5>Ouput 1: netMHCpan-ready input</h5>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="co">#display the netmhcpan-ready input / list of all peptides 9-12 aa, without </span></span>
<span id="cb15-2"><a href="#cb15-2"></a><span class="co">#modifications</span></span>
<span id="cb15-3"><a href="#cb15-3"></a><span class="kw">print</span>(<span class="kw">head</span>(results_step2_Human_Liver_AUTD17[[<span class="dv">1</span>]]))</span>
<span id="cb15-4"><a href="#cb15-4"></a><span class="co">#&gt;       Peptide</span></span>
<span id="cb15-5"><a href="#cb15-5"></a><span class="co">#&gt; 1   DYENLFLKF</span></span>
<span id="cb15-6"><a href="#cb15-6"></a><span class="co">#&gt; 13  RYFSTSVSW</span></span>
<span id="cb15-7"><a href="#cb15-7"></a><span class="co">#&gt; 17 AFSHLLLTTM</span></span>
<span id="cb15-8"><a href="#cb15-8"></a><span class="co">#&gt; 19  PYMARVAFF</span></span>
<span id="cb15-9"><a href="#cb15-9"></a><span class="co">#&gt; 21  LYRPTAAAF</span></span>
<span id="cb15-10"><a href="#cb15-10"></a><span class="co">#&gt; 31 FPVELAKYYM</span></span></code></pre></div>
</div>
<div id="ouput-2-database-search-results-updated" class="section level5">
<h5>Ouput 2: Database search results updated</h5>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co">#display the updated database search results table with the categorizations from </span></span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="co">#step1</span></span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="kw">print</span>(<span class="kw">head</span>(results_step2_Human_Liver_AUTD17[[<span class="dv">2</span>]]))</span>
<span id="cb16-4"><a href="#cb16-4"></a><span class="co">#&gt;     Peptide X.10lgP     Mass Length ppm      m.z Z   RT   Area Fraction    Id</span></span>
<span id="cb16-5"><a href="#cb16-5"></a><span class="co">#&gt; 1 DYENLFLKF   23.98 1187.586      9 0.1 594.8004 2 73.2 243590        3 44426</span></span>
<span id="cb16-6"><a href="#cb16-6"></a><span class="co">#&gt; 2 DYENLFLKF   23.72 1187.586      9 0.1 594.8004 2 73.2 243590        3 44427</span></span>
<span id="cb16-7"><a href="#cb16-7"></a><span class="co">#&gt; 3 DYENLFLKF   23.13 1187.586      9 0.1 594.8004 2 73.2 243590        3 44428</span></span>
<span id="cb16-8"><a href="#cb16-8"></a><span class="co">#&gt; 4 DYENLFLKF   22.98 1187.586      9 0.1 594.8004 2 73.2 243590        3 44429</span></span>
<span id="cb16-9"><a href="#cb16-9"></a><span class="co">#&gt; 5 DYENLFLKF   22.70 1187.586      9 0.1 594.8004 2 73.2 243590        3 44430</span></span>
<span id="cb16-10"><a href="#cb16-10"></a><span class="co">#&gt; 6 DYENLFLKF   20.88 1187.586      9 0.1 594.8004 2 73.2 243590        3 44432</span></span>
<span id="cb16-11"><a href="#cb16-11"></a><span class="co">#&gt;       Scan from.Chimera</span></span>
<span id="cb16-12"><a href="#cb16-12"></a><span class="co">#&gt; 1 F3:16451           No</span></span>
<span id="cb16-13"><a href="#cb16-13"></a><span class="co">#&gt; 2 F3:16501           No</span></span>
<span id="cb16-14"><a href="#cb16-14"></a><span class="co">#&gt; 3 F3:16546           No</span></span>
<span id="cb16-15"><a href="#cb16-15"></a><span class="co">#&gt; 4 F3:16609           No</span></span>
<span id="cb16-16"><a href="#cb16-16"></a><span class="co">#&gt; 5 F3:16633           No</span></span>
<span id="cb16-17"><a href="#cb16-17"></a><span class="co">#&gt; 6 F3:16686           No</span></span>
<span id="cb16-18"><a href="#cb16-18"></a><span class="co">#&gt;                                                 Source.File</span></span>
<span id="cb16-19"><a href="#cb16-19"></a><span class="co">#&gt; 1 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-20"><a href="#cb16-20"></a><span class="co">#&gt; 2 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-21"><a href="#cb16-21"></a><span class="co">#&gt; 3 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-22"><a href="#cb16-22"></a><span class="co">#&gt; 4 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-23"><a href="#cb16-23"></a><span class="co">#&gt; 5 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-24"><a href="#cb16-24"></a><span class="co">#&gt; 6 171002_AM_BD-ZH17_Liver_W_10%_DDA_#3_400-650mz_msms6.mzML</span></span>
<span id="cb16-25"><a href="#cb16-25"></a><span class="co">#&gt;                  Accession PTM AScore Found.By Peptide_no_mods</span></span>
<span id="cb16-26"><a href="#cb16-26"></a><span class="co">#&gt; 1 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb16-27"><a href="#cb16-27"></a><span class="co">#&gt; 2 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb16-28"><a href="#cb16-28"></a><span class="co">#&gt; 3 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb16-29"><a href="#cb16-29"></a><span class="co">#&gt; 4 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb16-30"><a href="#cb16-30"></a><span class="co">#&gt; 5 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb16-31"><a href="#cb16-31"></a><span class="co">#&gt; 6 |denovo_HF_fake_protein9            PEAKS DB       DYENLFLKF</span></span>
<span id="cb16-32"><a href="#cb16-32"></a><span class="co">#&gt;   Potential_spliceType</span></span>
<span id="cb16-33"><a href="#cb16-33"></a><span class="co">#&gt; 1               Linear</span></span>
<span id="cb16-34"><a href="#cb16-34"></a><span class="co">#&gt; 2               Linear</span></span>
<span id="cb16-35"><a href="#cb16-35"></a><span class="co">#&gt; 3               Linear</span></span>
<span id="cb16-36"><a href="#cb16-36"></a><span class="co">#&gt; 4               Linear</span></span>
<span id="cb16-37"><a href="#cb16-37"></a><span class="co">#&gt; 5               Linear</span></span>
<span id="cb16-38"><a href="#cb16-38"></a><span class="co">#&gt; 6               Linear</span></span></code></pre></div>
</div>
</div>
<div id="export-2" class="section level4">
<h4>Export</h4>
<p>If export is set to <code>TRUE</code> and a valid directory is provided in <code>export_dir</code>, then the results are exported .csv, .csv and csv format, respectively.</p>
<p>Even if the export parameters were not set at the beginning, the results returned can always be exported with the <code>export_step2_results</code> function as long as as the results obtained from the step2_wo_netMHCpan function are stored which is also indicated in the results_list parameter of the <code>export_step2_results</code> function.</p>
</div>
</div>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Faridi, P., Li, C., Ramarathinam, S. H., Vivian, J. P., Illing, P. T., Mifsud, N. A., Ayala, R., Song, J., Gearing, L. J., Hertzog, P. J., Ternette, N., Rossjohn, J., Croft, N. P., &amp; Purcell, A. W. (2018). A subset of HLA-I peptides are not genomically templated: Evidence for cis- and trans-spliced peptide ligands. Science Immunology, 3(28), eaar3947. 10.1126/sciimmunol.aar3947, <a href="https://immunology.sciencemag.org/content/3/28/eaar3947">Link</a></p>
<p>Hanada K, Yewdell JW, Yang JC. Immune recognition of a human renal cancer antigen through post-translational protein splicing. Nature. 2004 Jan 15;427(6971):252-6. DOI 10.1038/nature02240, <a href="https://www.nature.com/articles/nature02240">Link</a></p>
<p>Marcu A, Bichmann L, Kuchenbecker L, et al HLA Ligand Atlas: a benign reference of HLA-presented peptides to improve T-cell-based cancer immunotherapyJournal for ImmunoTherapy of Cancer 2021;9:e002071. 10.1136/jitc-2020-002071, <a href="https://jitc.bmj.com/content/9/4/e002071">Link</a></p>
<p>Birkir Reynisson, Bruno Alvarez, Sinu Paul, Bjoern Peters, Morten Nielsen, NetMHCpan-4.1 and NetMHCIIpan-4.0: improved predictions of MHC antigen presentation by concurrent motif deconvolution and integration of MS MHC eluted ligand data, Nucleic Acids Research, Volume 48, Issue W1, 02 July 2020, Pages W449–W454, 10.1093/nar/gkaa379, <a href="https://academic.oup.com/nar/article/48/W1/W449/5837056">Link</a></p>
<p>Jurtz V, Paul S, Andreatta M, Marcatili P, Peters B, Nielsen M. NetMHCpan-4.0: Improved Peptide-MHC Class I Interaction Predictions Integrating Eluted Ligand and Peptide Binding Affinity Data. J Immunol. 2017 Nov 1;199(9):3360-3368. Epub 2017 Oct 4. PMID: 28978689; PMCID: PMC5679736 10.4049/jimmunol.1700893, <a href="https://academic.oup.com/nar/article/48/W1/W449/5837056">Link</a></p>
<p>The UniProt Consortium, UniProt: the universal protein knowledgebase in 2021, Nucleic Acids Research, Volume 49, Issue D1, 8 January 2021, Pages D480–D489, 10.1093/nar/gkaa1100, <a href="https://academic.oup.com/nar/article/49/D1/D480/6006196">Link</a></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
